arborist/resource.go

```diff
@@ -141,6 +150,43 @@ func resourceWithPath(db *sqlx.DB, path string) (*ResourceFromQuery, error) {
 		LIMIT 1
 	`
 	err := db.Select(&resources, stmt, path)
+	if len(resources) == 0 {
+		// not found
+		return nil, nil
+	}
+	if err != nil {
+		// query broken
+		return nil, err
+	}
+	resource := resources[0]
+	return &resource, nil
+}
+
+// resourceWithPath looks up a resource matching the given path. The database
+// schema guarantees such a resource to be unique. Any error returned is
+// because of internal database failure.
+func resourceWithTag(db *sqlx.DB, tag string) (*ResourceFromQuery, error) {
+	resources := []ResourceFromQuery{}
+	stmt := `
+		SELECT
+			parent.id,
+			parent.name,
+			parent.path,
+			parent.tag,
+			parent.description,
+			array(
+				SELECT child.path
+				FROM resource AS child
+				WHERE child.path ~ (
+					CAST ((ltree2text(parent.path) || '.*{1}') AS lquery)
+				)
+			) AS subresources
+		FROM resource AS parent
+		WHERE parent.tag = $1
+		GROUP BY parent.id
+		LIMIT 1
```

Comment:
This is a curious question because this is new to me: why the `GROUP BY` and `LIMIT` here pls?

Reply:
Haha it's a good question, I think neither is actually necessary here. Will remove.